src/coreclr/gc/gcinterface.h

```diff
@@ -265,7 +265,7 @@ enum GCEventLevel
 // Event keywords corresponding to events that can be fired by the GC. These
 // numbers come from the ETW manifest itself - please make changes to this enum
 // if you add, remove, or change keyword sets that are used by the GC!
-enum GCEventKeyword
+enum GCEventKeyword : uint64_t
```

Comment:
This enum is used by the GC/EE interface. Changing the size of the enum to `uint64_t` would required revving `GC_INTERFACE_MAJOR_VERSION`

Reply:
I expect that @Maoni0 would prefer to avoid revving the GC interface major version and do this in some other way.

Reply:
@noahfalk @brianrob: the current implementation of the provider/keyword/verbosity status update in [EtwCallbackCommon](https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/eventtrace.cpp#L2400) casts the 64 bit received keyword into a 32 bit enum for the GC. Unfortunately, the AllocationSamplingKeyword is beyond this range.
I did not realize (thanks for pointing this out @jkotas) that it would require to change the signature of the GC interface to support it.
Any idea?

Reply:
So far, I'm not seeing something that I would recommend.  I can think of at least one way to do this, but I don't even want to say it because I think it could break us into eventual jail.  The idea would be to use a bit that is already used on the ETW keyword mask, but not sent to the GC because the GC doesn't currently use it.  The risk is that if the GC ever wanted to use said bit, then there would be a problem.

So far, revving the interface seems best to me.

Reply:
I think we could just do the filtering on the VM side in the GCToCLREventSink no?
https://github.com/dotnet/runtime/blob/f6701e58766c8556919bc92cf2f9b33b6bb2383b/src/coreclr/vm/gctoclreventsink.cpp#L163

Reply:
Yes, that would be fine.  I had been trying to avoid the interface call cost, but I think it will not be a big deal, and certainly more preferrable to changing the interface.

Reply:
By following the execution path of `GCHeapUtilities::RecordEventStateChange`, it should be possible to call a new `GCHeapUtilities::AllocationSampledEventState(bool enable)` based on the value of the received keyword. This would be non generic processing like what is done for triggering a full GC. 
On the GC side, a simple boolean field in `GCEventStatus` would keep track of the state and be exposed like in my current code via `GCEventEnabledAllocationSampled()`.
This would also allow enable/disable detection that does not seem possible today (once enabled, a provider/keyword/verbosity stays enabled) and... avoid the virtual call  ;^)